src/status_im/hardwallet/core.cljs

```diff
@@ -56,6 +56,21 @@
       (:keycard-pairing
        (find-multiaccount-by-keycard-instance-uid db instance-uid))))))
 
+(fx/defn listen-to-hardware-back-button
+  [{:keys [db]}]
+  (when-not (get-in db [:hardwallet :back-button-listener])
+    {:hardwallet/listen-to-hardware-back-button nil}))
+
+(fx/defn remove-listener-to-hardware-back-button
+  [{:keys [db]}]
+  (when-let [listener (get-in db [:hardwallet :back-button-listener])]
+    {:hardwallet/remove-listener-to-hardware-back-button listener}))
+
+(fx/defn on-add-listener-to-hardware-back-button
+  {:events [:hardwallet/add-listener-to-hardware-back-button]}
```

Comment:
could you explain why the hardware navigate back button needs to be hijacked? is it something that we might need in other places of the code base (for instance contacts that navigates back o chat instead of profile when opened from profile)? please write some docstrings

Reply:
It's needed only during keycard setup to prevent user from going back after card state has already changed

Reply:
please add that explanation as a comment in the code